### DISCUSSION TITLE
How to change color of Button depending on variable in ViewModel?

### DISCUSSION THREAD

**Discussion by SwiftExtender** [AvaloniaUI/Avalonia#17318] [General]
I have button in app, which changes color depending on async task in viewmodel (is microphone working - one color, not working microphone - another color) and i cant understand what is the best way to do this without breaking MVVM pattern.
Is reading ViewModel variables in Avalonia Behaviors acceptable by MVVM?

**Comment by thevortexcloud** [comment]
The simplest way is to just use a class style binding.

https://docs.avaloniaui.net/docs/guides/data-binding/binding-classes

Otherwise you can also use a custom template selector in some situations.

https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/DataTemplates/IDataTemplateSample

**Comment by kyuranger** [comment]
There is a simplest way by add an enum to your viewmodel:
```
public enum StatusEnum {Working,Stop};
public StatusEnum Status{get;set;}
```

Then bind this property and use a converter to it, just like this:
`<Button Background="{Bind Status,Converter={StaticResources StatusToColorConverter}}"></Button>`

Of course, you have to write StatusToColorConverter yourself. I'm too lazy to write in detail, but the idea is roughly like this.

**Comment by SwiftExtender** [comment]
Big thanks for repliers. Class style binding is the easiest choice.
